roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,698 @@
+
+                docker_unit = self.read_command_output(['systemctl', 'cat',
+                                                        'docker.service'])
+                self.register_file('docker-unit-file', docker_unit)
+
+                # Save docker service unit environment file for openshift-sdn,
+                # if any.
+                #
+                # The systemd and service modules both return the unit as a
+                # dict with only the last value for a given key, and there may
+                # be multiple EnvironmentFile stanzas, so we need to slurp and
+                # parse the service unit without the aid of these two modules.
+                #
+                # XXX: Do we really need to look for an 'openshift-sdn'
+                # environment file? It is not evident that the installer ever
+                # configures one.
+                #
+                # TODO: Submit a PR to fix the systemd module and maybe the
+                # service module as well to return a list value for any key
+                # that is repeated in the unit.
+                #
+                # TODO: Submit a PR to fix the ConfigParser Python module to
+                # support repeated keys
+                # (cf. <https://bugs.python.org/issue31528>).
+                for line in docker_unit.splitlines():
+                    if '=' not in line:
+                        continue
+
+                    key, value = line.split('=', 1)
+                    if key == 'EnvironmentFile' and 'openshift-sdn' in value:
+                        self.register_file('docker-network-file', value)
+            except OpenShiftCheckException as exc:
+                self.register_failure(exc)
+
+        self.process_pods_for_node(node_name)
+
+        # Choose a pod to use as the base pod for connectivity checks.
+        base_pod = None
+        for pod in self.get_resource('pods'):
+            if pod['spec'].get('nodeName') != node_name:
+                continue
+
+            if pod.get('ovs_bridge_port', None) is None or \
+               pod.get('vnid', None) is None or \
+               pod.get('mac_addr', None) is None:
+                continue
+
+            if not base_pod or \
+               base_pod.get('metadata').get('namespace') == 'default':
+                base_pod = pod
+        if not base_pod:
+            self.register_failure(
+                'No pods on node %s, so no connectivity tests' % node_name)
+            return
+
+        self.check_node_pod_to_pod(base_pod)
+        self.check_pod_external_connectivity(base_pod)
+        self.check_node_pod_to_service(base_pod)
+
+    # def save_service_logs(self, service_name):
+    #     """Save the last 1 MiB of logs for the specified systemd unit."""
+    #     self.save_command_output('%s.log' % service_name,
+    #                              ' '.join(['journalctl', '-u', service_name,
```

Comment:
Can't we do journalctl -n 10000 and assume it will be about 1M?

It would be nice to make the -n quantity be a parameter, and also allow a --since option too.